lib/structs/email_params.py

```diff
@@ -1,28 +1,74 @@
-from dataclasses import dataclass
-from typing import List
+from dataclasses import dataclass, fields
+from typing import Optional, Dict, List
+from custom_types.email_api.aliases import (
+    TemplateMappings,
+    EmailAddress,
+    EmailAddresses,
+)
+from email_api.template_handler import TemplateHandler
 
 
 @dataclass
 # pylint:disable=too-many-instance-attributes
 class EmailParams:
     """
-    Email parameters that are shared between the send_email and send_emails functions in EmailApi
+    Email parameters that hold config info on how to structure and send an email
     :param: subject: (String): Subject of the email
     :param: email_from (String): Sender Email, subject (String): Email Subject,
-    :param: email_cc (List[String]): Email addresses to Cc
-    :param: header (String): filepath to header file,
-    :param: footer (String): filepath to footer file,
-    :param: attachment_filepaths (List): list of attachment filepaths,
-    :param: send_as_html (Bool): If true will send in HTML format
-    :return: (Status (Bool), Output <*>): tuple of action status (succeeded(T)/failed(F)) and the output
+    :param: email_cc (Tuple[String]): An Optional tuple of additional email addresses to Cc
+    :param: body_html (String): Email body (HTML)
+    :param: body_plaintext (String): Email body as plaintext
+    :param: attachment_filepaths: List(Strings): An Optional list of filepaths to files to attach to email
+        -must be relative filepaths which start from .../st2-cloud-pack/email_attachments
     """
 
     subject: str
-    email_from: str
-    email_cc: List[str]
-    header: str
-    footer: str
-    attachment_filepaths: List[str]
-    test_override: bool
-    test_override_email: List[str]
-    send_as_html: bool
+    email_from: EmailAddress
+    body_html: str
+    body_plaintext: str
+    email_cc: Optional[EmailAddresses] = None
+    attachment_filepaths: Optional[List[str]] = None
+
+    @staticmethod
+    def from_dict(dictionary: Dict):
+        """
+        Method which returns instance of this dataclass from a dictionary (when creating from template mappings)
+        :param dictionary: dictionary to use to instantiate dataclass
+        """
+        field_set = {field.name for field in fields(EmailParams) if field.init}
+        filtered_arg_dict = {
+            key: value for key, value in dictionary.items() if key in field_set
+        }
+        return EmailParams(**filtered_arg_dict)
+
+    @staticmethod
+    def from_template_mappings(template_mappings: TemplateMappings, **kwargs):

```

Comment:
Since we have multiple from_dict helpers now, I think it's worth bringing in a lib such as https://github.com/Fatal1ty/mashumaro to handle this for us